arches/app/etl_modules/base_import_module.py

```diff
@@ -367,27 +370,147 @@ def start(self, request):
             files = details["result"]["summary"]["files"]
             summary = details["result"]["summary"]
-            use_celery_file_size_threshold_in_MB = 0.1
-            if (
-                summary["cumulative_files_size"] / 1000000
-                > use_celery_file_size_threshold_in_MB
-            ):
+            use_celery_file_size_threshold = self.config.get(
+                "celeryByteSizeLimit", 100000
+            )
+
+            if summary["cumulative_files_size"] > use_celery_file_size_threshold:
                 response = self.run_load_task_async(request, self.loadid)
             else:
                 response = self.run_load_task(
-                    self.userid, files, summary, result, self.temp_dir, self.loadid
+                    self.userid,
+                    files,
+                    summary,
+                    result,
+                    self.temp_dir,
+                    self.loadid,
+                    multiprocessing,
                 )
 
             return response
 
+    def cli(self, source):
+        initiated = self.start(self.request)
+
+        if initiated["success"]:
+            read = self.read_for_cli(source)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": initiated["data"]["message"]},
+            }
+
+        if read["success"]:
+            self.request.POST.__setitem__(
+                "load_details", json.dumps({"result": read["data"]})
+            )
+            self.request.POST.__setitem__("multiprocessing", True)
+            written = self.write(self.request)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": read["data"]["message"]},
+            }
+
+        if written["success"]:
+            return {"success": True, "data": "done"}
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": written["data"]["message"]},
+            }
+
+    def read_for_cli(self, source):
+        self.cumulative_files_size = 0
+        self.temp_dir = os.path.join(settings.UPLOADED_FILES_DIR, "tmp", self.loadid)
+        try:
+            self.delete_from_default_storage(self.temp_dir)
+        except FileNotFoundError:
+            pass
+        file_name = os.path.basename(source).split("/")[-1]
+        file_stat = os.stat(source)
+
+        result = {
+            "summary": {
+                "name": file_name,
+                "size": self.filesize_format(file_stat.st_size),
+                "files": {},
+            }
+        }
+        validator = FileValidator()
+        if len(validator.validate_file_type(source)) > 0:
+            return {
+                "status": 400,
+                "success": False,
+                "title": _("Invalid excel file/zip specified"),
+                "message": _("Upload a valid excel file"),
```

Comment:
Could this message be "Upload a valid .xlsx or .zip file"? I think we don't accept .xls files, so it might be better to explicitly state `.xlsx` rather than 'excel'.